internal/file/file_manager_service.go

```diff
@@ -521,3 +550,98 @@ func (fms *FileManagerService) UpdateCurrentFilesOnDisk(currentFiles map[string]
 		fms.currentFilesOnDisk[file.GetFileMeta().GetName()] = file
 	}
 }
+
+func (fms *FileManagerService) UpdateManifestFile(currentFiles map[string]*mpi.File) (err error) {
+	manifestFiles := fms.convertToManifestFileMap(currentFiles)
+	manifestJSON, err := json.MarshalIndent(manifestFiles, "", "  ")
+	if err != nil {
+		return fmt.Errorf("unable to marshal manifest file json: %w", err)
+	}
+
+	// 0755 allows read/execute for all, write for owner
+	if err = os.MkdirAll(manifestDirPath, dirPerm); err != nil {
+		return fmt.Errorf("unable to create directory %s: %w", manifestDirPath, err)
+	}
+
+	// 0600 ensures only root can read/write
+	newFile, err := os.OpenFile(manifestFilePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, filePerm)
+	if err != nil {
+		return fmt.Errorf("failed to read manifest file: %w", err)
+	}
+	defer newFile.Close()
+
+	_, err = newFile.Write(manifestJSON)
+	if err != nil {
+		return fmt.Errorf("failed to write manifest file: %w", err)
+	}
+
+	return nil
+}
+
+func (fms *FileManagerService) getManifestFile(currentFiles map[string]*mpi.File) (map[string]*mpi.File, error) {
```

Comment:
Could this be renamed 
```suggestion
func (fms *FileManagerService) manifestFile(currentFiles map[string]*mpi.File) (map[string]*mpi.File, error) {
```

Reply:
I found `getManifestFile` to be more descriptive. But can rename it if that's more cleaner?